subsys/net/ip/ipv6.c

```diff
@@ -222,12 +222,7 @@ struct net_ipv6_nbr_data *net_ipv6_get_nbr_by_index(u8_t idx)
 
 static inline void nbr_clear_ns_pending(struct net_ipv6_nbr_data *data)
 {
-	int ret;
-
-	ret = k_delayed_work_cancel(&data->send_ns);
-	if (ret < 0) {
-		NET_DBG("Cannot cancel NS work (%d)", ret);
-	}
+	k_delayed_work_cancel(&data->send_ns);
```

Comment:
Most user of k_delayed_work_cancel don't check the return, it is only really useful to tell the difference if the work has run or not but if we all we care is that it don't get executed that is now guaranteed regardless of the return.